Samples/iOS-ObjectiveC/iOS-ObjectiveC/AppDelegate.m

```diff
@@ -33,39 +33,32 @@ - (BOOL)application:(UIApplication *)application
             options.tracesSampleRate = @([env[@"--io.sentry.tracesSampleRate"] doubleValue]);
         }
 
-        if ([args containsObject:@"--io.sentry.profile-options-v2"]) {
-            options.configureProfiling = ^(SentryProfileOptions *_Nonnull profiling) {
-                profiling.lifecycle = [args containsObject:@"--io.sentry.profile-lifecycle-trace"]
-                    ? SentryProfileLifecycleTrace
-                    : SentryProfileLifecycleManual;
-
-                profiling.sessionSampleRate = 1.f;
-                if (env[@"--io.sentry.profile-session-sample-rate"] != nil) {
-                    profiling.sessionSampleRate =
-                        [env[@"--io.sentry.profile-session-sample-rate"] floatValue];
-                }
-            };
-        } else {
-            NSNumber *profilesSampleRate = @1;
-            if ([args containsObject:@"--io.sentry.enableContinuousProfiling"]) {
-                profilesSampleRate = nil;
-            } else if (env[@"--io.sentry.profilesSampleRate"] != nil) {
-                profilesSampleRate = @([env[@"--io.sentry.profilesSampleRate"] floatValue]);
-            }
-            options.profilesSampleRate = profilesSampleRate;
-
-            if (![args containsObject:@"--io.sentry.enableContinuousProfiling"]
-                && env[@"--io.sentry.profilesSamplerValue"] != nil) {
-                options.profilesSampler
-                    = ^NSNumber *_Nullable(SentrySamplingContext *_Nonnull samplingContext)
-                {
-                    return @([env[@"--io.sentry.profilesSamplerValue"] floatValue]);
-                };
+        options.configureProfiling = ^(SentryProfileOptions *_Nonnull profiling) {
+            profiling.lifecycle = [args containsObject:@"--io.sentry.profile-lifecycle-manual"]
+                ? SentryProfileLifecycleManual
+                : SentryProfileLifecycleTrace;
+
+            profiling.sessionSampleRate = 1.f;
+            if (env[@"--io.sentry.profile-session-sample-rate"] != nil) {
+                profiling.sessionSampleRate =
+                    [env[@"--io.sentry.profile-session-sample-rate"] floatValue];
             }
+        };
 
-            if ([args containsObject:@"--io.sentry.profile-app-launches"]) {
-                options.enableAppLaunchProfiling = YES;
-            }
+        if (env[@"--io.sentry.profilesSampleRate"] != nil) {
+            options.profilesSampleRate = @([env[@"--io.sentry.profilesSampleRate"] floatValue]);
+        }
+
+        if (env[@"--io.sentry.profilesSamplerValue"] != nil) {
+            options.profilesSampler
+                = ^NSNumber *_Nullable(SentrySamplingContext *_Nonnull samplingContext)
+            {
+                return @([env[@"--io.sentry.profilesSamplerValue"] floatValue]);
+            };
+        }
+
+        if ([args containsObject:@"--io.sentry.disable-app-start-profiling"]) {
+            options.enableAppLaunchProfiling = YES;
```

Comment:
This looks flipped, the option says disable but you're enabling it

Reply:
+1

Seems that you want `if (![args containsObject...`

Reply:
yep, forgot to flip it after changing the sample app's default setting. thanks!